Belch AI Scoring Fix

Pawkkie opened the pull request:
## Description
@Kasen discovered today that Belch was being considered an invalid option (score and damage of 0) by the AI even after it had eaten a berry. While exploring the issue I discovered that the check used to determine whether a berry has been eaten was completely different in `IsDamageMoveUnusable` and `IsBelchPreventingMove`, with the latter implementation being significantly older. I updated it to match the newer implementation, and it fixed the issue.

I wrote a test that builds a situation in which the AI should use Belch, which fails on current master but passes with this PR. I also moved the only other Belch test I could find into its own file with this one as there are now two Belch tests, though one relates to the AI and the other was in with Stuff Cheeks so if you'd like a different organization let me know.

## Images
Sample scenario before this fix, from the turn after eating a berry:
![image](https://github.com/rh-hideout/pokeemerald-expansion/assets/61265402/34b8d7b2-b225-48ec-b504-3eff21e9c323)

Same scenario with this fix:
![image (1)](https://github.com/rh-hideout/pokeemerald-expansion/assets/61265402/d69696c6-5c00-485e-b4db-fd3634d65d72)


## **People who collaborated with me in this PR**
@Kasen found the issue, I've written the code!

## **Discord contact info**
@Pawkkie

AlexOn1ine commented:
> I updated it to match the newer implementation, and it fixed the issue
My suspicion is that the older check is correct and the newer wrong (that was added by me). Can you instead try and adjust the new on please?

Pawkkie commented:
> > I updated it to match the newer implementation, and it fixed the issue
> 
> My suspicion is that the older check is correct and the newer wrong (that was added by me). Can you instead try and adjust the new on please?

I just tried this out, replacing the newer one in `IsDamageMoveUnusable` with the older version, and the test fails. Maybe I've implemented it wrong or maybe the problem is elsewhere, not sure, but the AI only seems to see the move as usable with the `IsBelchPreventingMove` function changed to match your version.

![image](https://github.com/rh-hideout/pokeemerald-expansion/assets/61265402/c3f7e943-7a2f-4586-9a57-32538b93eff7)

Cases like Knock Off or item theft etc. are definitely not accounted for in this. I'm not sure if there's a good way to blanket catch all of those cases without needing to manually specify them, which is a bit awkward for futureproofing. Would you prefer I instead make an issue for this bug and just PR the new test marked as "known failing"? I have just updated the new test to be AI_SINGLE_BATTLE_TEST, completely forgot that was a thing the first time lol

AlexOn1ine commented:
> Cases like Knock Off or item theft etc. are definitely not accounted for in this
If the ateBerry check in battle_util.c is removed it will be a bug, not an accuracy issue. 

I would just keep this PR open and I can take look later and maybe help a bit out with solving this.

Pawkkie commented:
Yeah sure! No rush, just wanted to get this written up while it was fresh, which I'm trying to be better at these days lol

AlexOn1ine commented:
@Pawkkie could you make the modifications above and additionally these:
```diff
--- a/src/battle_script_commands.c
+++ b/src/battle_script_commands.c
@@ -2115,6 +2115,7 @@ static void Cmd_multihitresultmessage(void)
     if (gSpecialStatuses[gBattlerTarget].berryReduced
         && !(gMoveResultFlags & MOVE_RESULT_NO_EFFECT))
     {
+        gBattleStruct->ateBerry[gBattlerTarget & BIT_SIDE] |= gBitTable[gBattlerPartyIndexes[gBattlerTarget]];
         gSpecialStatuses[gBattlerTarget].berryReduced = FALSE;
         BattleScriptPushCursor();
         gBattlescriptCurrInstr = BattleScript_PrintBerryReduceString;
@@ -2575,6 +2576,7 @@ static void Cmd_resultmessage(void)
     if (gSpecialStatuses[gBattlerTarget].berryReduced
         && !(gMoveResultFlags & MOVE_RESULT_NO_EFFECT))
     {
+        gBattleStruct->ateBerry[gBattlerTarget & BIT_SIDE] |= gBitTable[gBattlerPartyIndexes[gBattlerTarget]];
         gSpecialStatuses[gBattlerTarget].berryReduced = FALSE;
         BattleScriptPushCursor();
         gBattlescriptCurrInstr = BattleScript_PrintBerryReduceString;
 }
```
The test should pass after this. It did for me.  

Btw if you used an Oran or Sitrus Berry instead we wouldn't be able to catch this. But this is also a very niche interaction so no wonder nobody caught it yet.

AlexOn1ine commented on src/battle_util.c:
```diff
@@ -1262,12 +1262,15 @@ bool32 IsHealBlockPreventingMove(u32 battler, u32 move)
         return FALSE;
 
-    return !(gBattleStruct->ateBerry[battler & BIT_SIDE] & gBitTable[gBattlerPartyIndexes[battler]]);
+    if (ItemId_GetPocket(GetUsedHeldItem(battler)) != POCKET_BERRIES)
+        return TRUE;
+    else
+        return FALSE;
```
this one needs to be reverted
